### PR TITLE
Add date_taken to photos

### LINKED ISSUE #1421: Order photos by date taken
currently we order by the local database :created_at, but both flickr and the jpeg have a date taken field. When the user imports several photos of a plant, we can order by the date the photo was taken, and then it appears to grow from photo to photo.

image

### PR DESCRIPTION
**Pull request by mftaff**
closes #1421

### CODE REVIEW

**Comment by Br3nda** [comment]
Looks good. I think you forgot to check in `schema.rb`

**Comment by mftaff** [comment]
Whoops..
Just added it now

File: app/controllers/plantings_controller.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def show
     @planting = Planting.includes(:owner, :crop, :garden, :photos)
       .friendly
       .find(params[:id])
-    @photos = @planting.photos.order(created_at: :desc).includes(:owner).paginate(page: params[:page])
+    @photos = @planting.photos.order(date_taken: :asc).includes(:owner).paginate(page: params[:page])
```
Comment (Br3nda) [inline app/controllers/plantings_controller.rb]: Can we keep it as descending order? Newest photos first.

Comment (mftaff) [reply app/controllers/plantings_controller.rb]: for sure! 
changing now.

File: app/controllers/plantings_controller.rb [hunk 2]
```diff
@@ -31,7 +31,7 @@ def show
     @planting = Planting.includes(:owner, :crop, :garden, :photos)
       .friendly
       .find(params[:id])
-    @photos = @planting.photos.order(created_at: :desc).includes(:owner).paginate(page: params[:page])
+    @photos = @planting.photos.order(date_taken: :desc).includes(:owner).paginate(page: params[:page])
```
Comment (CloCkWeRX) [inline app/controllers/plantings_controller.rb]: Do we want to migrate old data (fall back on created at if null? Fetch from flickr again?)

Comment (Br3nda) [reply app/controllers/plantings_controller.rb]: What's the performance hit of that?


ALSO, i just realised we'd want to index that column too - so the query can order it faster

Comment (CloCkWeRX) [reply app/controllers/plantings_controller.rb]: Updating existing records in a migration to fall back to created_at or re-lookup from flickr - one off, no future perf considerations

Using COALESCE... its fairly fast, particularly if we added the index on the coalesce key itself.

**Comment by Br3nda** [comment]
@mftaff Thank you for your work here. I rebased the dev branch and it's made a mess of your PR. I'll fix this up for you, probably later today.

**Comment by Br3nda** [comment]
Replaced with #1568

**Pull request closed without merging**